admin/class-acc_user_importer-admin.php

```diff
@@ -356,94 +301,58 @@ private function proccess_user_data ( $users ) {
 			$user_id = username_exists($userContactId);
-			$update_this_user = false;
-			
+			$updatedFields = [];
+
+			// Existing user, check if any fields were updated
 			if( is_numeric( $user_id ) ) {
 				//---------USER WAS FOUND IN DATABASE------------
 				$this->log_dual(" > found " . $userContactId . " (user #" . $user_id . ")");
 
 				//append ID to data object - wordpress will update the user instead of create a new one
-				$user_data = array_merge( array('ID' => $user_id), $user_data);
+				$accUserData = array_merge( array('ID' => $user_id), $accUserData);
 
 				//Update DB only if something changed.
 				//Do a bunch of compare to decide if an update is needed.
 				//This logic needs to be extended each time we add a new user field.
 				// Get the user object
 				$user_meta = get_userdata($user_id);
 
-				// Get all the user roles as an array.
-				$user_roles = $user_meta->roles;
-				// Check if user has expired or ex-member role. If so, update role to default
-				if ( in_array( $role_expiry1, $user_roles, true ) ) {
-					$this->log_dual(" > User is " . $role_expiry1 . ", refreshing role to " . $default_role);
-					$update_this_user = true;
-					$user_data["role"] = $default_role;
-					$role_refreshed[] = $userContactId;
-					$role_refreshed_email[] = $userEmail;
-				} elseif (in_array( $role_expiry2, $user_roles, true ) ) {
-					$this->log_dual(" > User is " . $role_expiry2 . ", refreshing role to " . $default_role);
-					$update_this_user = true;
-					$user_data["role"] = $default_role;
-					$role_refreshed[] = $userContactId;
-					$role_refreshed_email[] = $userEmail;
-				}
-
-				//Check if email changed
-				if ($userEmail != $user_meta->user_email) {
-					$this->log_dual(" > email changed from " . $user_meta->user_email . " to " . $userEmail);
-					$update_this_user = true;
-				}
-
-				//Check if HomePhone changed
-				if ($userHomePhone != $user_meta->home_phone) {
-					$this->log_dual(" > home phone changed from " . $user_meta->home_phone . " to " . $userHomePhone);
-					$update_this_user = true;
-				}
-
-				//Check if Cell phone changed
-				if ($userCellPhone != $user_meta->cell_phone) {
-					$this->log_dual(" > cell phone changed from " . $user_meta->cell_phone . " to " . $userCellPhone);
-					$update_this_user = true;
-				}
-
-				//Check if MEMBERSHIP_N changed
-				if ($userMembership != $user_meta->membership) {
-					$this->log_dual(" > membership# changed from " . $user_meta->membership . " to " . $userMembership);
-					$update_this_user = true;
-				}
-
-				//Check if Membership Expiry changed
-				if ($userExpiry != $user_meta->expiry) {
-					$this->log_dual(" > expiry changed from " . $user_meta->expiry . " to " . $userExpiry);
-					$update_this_user = true;
-				}
-
-				//Check if city changed
-				if ($userCity != $user_meta->city) {
-					$this->log_dual(" > city changed from " . $user_meta->city . " to " . $userCity);
-					$update_this_user = true;
-				}
 
```

Comment:
Looks like you are a more advanced PHP coder than me, lol.  You replaced all those checks with an array, good!